cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -88,13 +88,14 @@ class cufile_shim {
 void cufile_shim::modify_cufile_json() const
 {
   std::string const json_path_env_var = "CUFILE_ENV_PATH_JSON";
-  temp_directory tmp_config_dir{"cudf_cufile_config"};
+  static std::optional<temp_directory> tmp_config_dir;
+  if (not tmp_config_dir.has_value()) { tmp_config_dir = temp_directory{"cudf_cufile_config"}; }
```

Comment:
Could the static variable be initialized like this instead?
```suggestion
  static temp_directory tmp_config_dir = temp_directory{"cudf_cufile_config"};
```

Reply:
It can, I don't know what I was thinking here :D